### ISSUE TITLE
CI: Unable to ping endpoint with init policy. TTL exceeded

### ISSUE THREAD

**Issue by aanm** [cilium/cilium#5322]
```
20:24:14     Cannot ping endpoint with init policy
20:24:14     Expected command: ping -W 2 -c 5 10.15.93.97 
20:24:14     To succeed, but it failed:
20:24:14     Exitcode: 1 
20:24:14     Stdout:
20:24:14      	 PING 10.15.93.97 (10.15.93.97) 56(84) bytes of data.
20:24:14     	 From 10.15.0.1 icmp_seq=1 Time to live exceeded
20:24:14     	 From 10.15.0.1: icmp_seq=1 Redirect Host(New nexthop: 10.15.0.1)
20:24:14     	 From 10.15.0.1 icmp_seq=2 Time to live exceeded
20:24:14     	 From 10.15.0.1: icmp_seq=2 Redirect Host(New nexthop: 10.15.0.1)
20:24:14     	 From 10.15.0.1: icmp_seq=3 Redirect Host(New nexthop: 10.15.0.1)
20:24:14     	 From 10.15.0.1 icmp_seq=3 Time to live exceeded
20:24:14     	 From 10.15.0.1: icmp_seq=4 Redirect Host(New nexthop: 10.15.0.1)
20:24:14     	 From 10.15.0.1 icmp_seq=4 Time to live exceeded
20:24:14     	 From 10.15.0.1 icmp_seq=5 Time to live exceeded
20:24:14     	 
20:24:14     	 --- 10.15.93.97 ping statistics ---
20:24:14     	 5 packets transmitted, 0 received, +5 errors, 100% packet loss, time 4015ms
20:24:14     	 
20:24:14     	 
20:24:14     Stderr:
```
https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/1461/execution/node/96/log/?consoleFull

[616c54a2_RuntimePolicies_Init_Policy_Test_Init_Ingress_Policy_Test.zip](https://github.com/cilium/cilium/files/2315272/616c54a2_RuntimePolicies_Init_Policy_Test_Init_Ingress_Policy_Test.zip)

**Comment by aanm** [comment]
Discussion with @joestringer we think it's because the endpoint's entry in the lxcmap is not written after the endpoint add API request is completed.

**Comment by joestringer** [comment]
Is the endpoint add API request always synchronous? I think this was changed for the CNI plugin some time ago, I wonder if this was changed in the core API code or if there is some difference in the client request which needs to also occur from the docker plugin.

**Comment by joestringer** [comment]
@aanm how frequently do we hit this?

**Comment by aanm** [comment]
@joestringer so far only one time.

**Comment by joestringer** [comment]
I believe that in this scenario, we don't know how to deliver to the destination because the endpoint no longer exists on the local node. Even if we handled this more gracefully, I think it would involve explicitly checking this condition and dropping the packet rather than forwarding on to another device which loops the packet back and ends up dropping the packet by looping and dropping TTL until the packet expires.

**Comment by ianvernon** [comment]
Can this be closed? We haven't hit this in some time.

**Comment by tgraf** [comment]
This was fixed by #5663
